optuna/testing/storages.py

```diff
@@ -90,7 +89,7 @@ def __enter__(
         elif self.storage_specifier == "grpc":
             self.tempfile = NamedTemporaryFilePool().tempfile()
             url = "sqlite:///{}".format(self.tempfile.name)
-            port = 13000 + uuid.uuid4().int % 1000
+            port = self.extra_args.get("port", 13000)
```

Comment:
How about finding a free port here instead of updating `tests/study_tests/test_study.py`?

```python
$ python3
Python 3.11.7 (main, Jan 20 2024, 14:32:50) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>>> sock.getsockname()
('0.0.0.0', 0)
>>> sock.bind(('', 0))
>>> _, port = sock.getsockname()
>>> port
58157
```

Alternatively, you can find a free port within a specific range like this:

```python
def find_free_port():
    for port in range(13000, 13100):
        try:
            s.bind(('', port))
        except OSError:
            continue
    return port
```

Using one of these methods could simplify unit tests. What do you think?